lsp-completion.el

```diff
@@ -455,11 +454,52 @@ The MARKERS and PREFIX value will be attached to each candidate."
 
 (defun lsp-completion--get-documentation (item)
   "Get doc comment for completion ITEM."
-  (-some->> item
-    (lsp-completion--resolve)
-    (get-text-property 0 'lsp-completion-item)
-    (lsp:completion-item-documentation?)
-    (lsp--render-element)))
+  (-let* ((resolved (get-text-property 0 'lsp-completion-resolved item))
+          (item (get-text-property
+                 0
+                 (if resolved
+                     'lsp-completion-item
+                   'lsp-completion-unresolved-item)
+                 item))
+          ((&CompletionItem :detail?
+                            :documentation?)
+           item))
+
+    (unless (or resolved (and detail? documentation?))
+      (setq item (lsp-completion--resolve item)
+            resolved t))
+
+    (setq detail? (lsp:completion-item-detail? item)
+          documentation? (lsp:completion-item-documentation? item))
+
+    (when (and detail? documentation?)
+      (cond ((lsp-markup-content? documentation?)
+             (-let (((&MarkupContent :kind :value) documentation?))
+               (cond ((and (equal kind "plaintext")
+                           (not (string-match-p (regexp-quote detail?) value)))
+                      (lsp:set-markup-content-value
+                       documentation?
+                       (concat detail?
+                               (if (bound-and-true-p page-break-lines-mode)
+                                   "\n\n"
+                                 "\n\n")
+                               value)))
+                     ((and (equal kind "markdown")
+                           (not (string-match-p (regexp-quote detail?) value)))
```

Comment:
instead of modifying the `documentation?` property of completion item. Can we just cache it to the text instead? Something like `'lsp-completion-item-doc`. That would make us don't have to make this sub string match every time we need to display this document again.

Reply:
I've implemented a cache for this now.